### ISSUE TITLE
Build linux PV tools from sources

### ISSUE THREAD

**Issue by stormi** [xcp-ng/xcp#73]
The PV tools for linux are currently binaries extracted from XenServer's tools ISO, because the source RPM for the tools does not contain build instructions nor sources.

The sources are available on github, so it would be cleaner (and easier to maintain) to do the build ourselves.

Sources: https://github.com/xenserver/xe-guest-utilities/

**Comment by olivierlambert** [comment]
Also, it could be great to use this knowledge to provide main distro packages, so you need to `yum install xcp-ng-tools` (or `apt install xcp-ng-tools`) in your VM, instead of relying on some ISO to insert.

**Comment by borzel** [comment]
Ubuntu has it already:
```
apt search xe-guest-utilities
[...]
xe-guest-utilities/bionic 7.10.0-0ubuntu1 i386
  daemon for monitoring Xen Virtual machines
```

**Comment by olivierlambert** [comment]
Didn't know that :) This could be useful as a "template" to get some work for CentOS/Debian/other distro without it?

**Comment by stormi** [comment]
Yes, it could be an inspiration

**Comment by Nambrok** [comment]
I use `xe-guest-utilities-git` from [AUR](https://aur.archlinux.org/packages/xe-guest-utilities-git/) on my ArchLinux VM too.

I just modified it to build from XCP-ng repo.

**Comment by stormi** [comment]
Looks like you could contribute a nice addition to the docs here: https://xcp-ng.org/docs/guests.html#install-from-the-distro-s-online-repositories

**Comment by stormi** [comment]
Fixed by https://github.com/xcp-ng-rpms/xcp-ng-pv-tools/commit/3d134fbd0f3ac9e2f3cfa914e38b33529635d458 and released recently to 8.2 (see https://xcp-ng.org/blog/2021/06/28/summer-security-and-bugfix-updates/)
